src/api/todos.ts

```diff
@@ -0,0 +1,54 @@
+import { Todo } from '../types/Todo';
+import { client } from '../utils/fetchClient';
+
+export const USER_ID = 292;
+
+export const getTodos = () => {
+  return client.get<Todo[]>(`/todos?userId=${USER_ID}`);
+};
+
+// Add more methods here
+export const getCount = (todos: Todo[]): number => {
+  if (todos) {
+    return todos.filter(todo => !todo.completed).length;
+  }
+
+  return 0;
+};
+
+export const addTodo = (postId: number, postTitle: string) => {
+  return client.post<Todo>(`/todos`, {
+    id: postId,
+    userId: 292,
+    title: postTitle,
+    completed: false,
+  });
+};
+
+export const deleteTodo = (todoId: number) => {
+  return client.delete(`/todos/${todoId}`);
+};
+
+export const handleDeleteTodo = (
+  todo: Todo,
+  todoId: number,
+  setIsLoading: (e: number | null) => void,
+  setErrorMessage: (m: string) => void,
+) => {
+  setIsLoading(todoId);
+  deleteTodo(todo.id)
```

Comment:
same

Reply:
Also, you need to make the same loading logic on every API function. And better have just API function like addTodo
and create this handler in component where you use this function